spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,427 @@
+
+  public static final ConfigOption<Boolean> JOSDKMetricsEnabled =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.josdkMetricsEnabled")
+          .description(
+              "When enabled, the josdk metrics will be added in metrics source and "
+                  + "configured for operator.")
+          .defaultValue(true)
+          .build();
+
+  public static final ConfigOption<Boolean> KubernetesClientMetricsEnabled =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.clientMetricsEnabled")
+          .defaultValue(true)
+          .description(
+              "Enable KubernetesClient metrics for measuring the HTTP traffic to "
+                  + "the Kubernetes API Server. Since the metrics is collected "
+                  + "via Okhttp interceptors, can be disabled when opt in "
+                  + "customized interceptors.")
+          .build();
+
+  public static final ConfigOption<Boolean> KubernetesClientMetricsGroupByResponseCodeGroupEnabled =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.clientMetricsGroupByResponseCodeEnabled")
+          .description(
+              "When enabled, additional metrics group by http response code group(1xx, "
+                  + "2xx, 3xx, 4xx, 5xx) received from API server will be added. Users "
+                  + "can disable it when their monitoring system can combine lower level "
+                  + "kubernetes.client.http.response.<3-digit-response-code> metrics.")
+          .defaultValue(true)
+          .build();
+  public static final ConfigOption<Integer> OperatorMetricsPort =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.metrics.port")
+          .defaultValue(19090)
+          .description("The port used for checking metrics")
+          .typeParameterClass(Integer.class)
+          .enableDynamicOverride(false)
+          .build();
+  public static final ConfigOption<Integer> OperatorProbePort =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.probePort")
+          .defaultValue(18080)
+          .description("The port used for health/readiness check probe status.")
+          .typeParameterClass(Integer.class)
+          .enableDynamicOverride(false)
+          .build();
+  public static final ConfigOption<Integer> SentinelExecutorServicePoolSize =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.sentinelExecutorPoolSize")
+          .description(
+              "Size of executor service in Sentinel Managers to check the health "
+                  + "of sentinel resources.")
+          .defaultValue(3)
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .build();
+
+  public static final ConfigOption<Long> SENTINEL_RESOURCE_RECONCILIATION_DELAY =
+      ConfigOption.<Long>builder()
```

Comment:
Just a question. May I ask why this is `Long` type in seconds? Is it reasonable?

Reply:
Do we support 68 years as delay?
```
jshell> Integer.MAX_VALUE/60/60/24/365
$1 ==> 68
```

Reply:
Not quite - it's made a long mostly because we use a [ScheduledExecutorService](https://docs.oracle.com/en/java/javase/17/docs//api/java.base/java/util/concurrent/ScheduledExecutorService.html) to perform the check, and it takes a long for delay. We can definitely do a int -> long conversion if needed.